### PR TITLE
Feature: SKG-1 Forgot Password - User sets up a new password via rese…

### PR DESCRIPTION
**Pull request by KaranGupta-git**
…t link received.

### CODE REVIEW

File: src/app/modules/auth/components/reset-password/reset-password.component.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    this.matcher = new CustomErrorStateMatcher();
+  }
+
+  private createRouteSubscription() {
+    this.routeSubscription = this.activatedRoute.params.subscribe((params) => {
+      this.compositeField = params['uid'] ? params['uid'] : null;
+    });
+  }
+
+  private checkPasswords(group: FormGroup) {
```
Comment (vinishaaws) [inline src/app/modules/auth/components/reset-password/reset-password.component.ts]: we can reuse this if its duplicate code

File: src/app/modules/auth/components/reset-password/reset-password.component.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  ) {}
+
+  ngOnInit() {
+    this.createRouteSubscription();
+    this.initializeResetPasswordForm();
+    this.matcher = new CustomErrorStateMatcher();
+  }
+
+  private createRouteSubscription() {
+    this.routeSubscription = this.activatedRoute.params.subscribe((params) => {
```
Comment (vinishaaws) [inline src/app/modules/auth/components/reset-password/reset-password.component.ts:36]: what are you getting as uid? user Id or email / phone number?

File: src/app/modules/auth/components/reset-password/reset-password.component.ts (hunk 1 above)
Comment (KaranGupta-git) [reply src/app/modules/auth/components/reset-password/reset-password.component.ts]: ok, I am moving it to helper

File: src/app/modules/auth/components/reset-password/reset-password.component.ts (hunk 2 above)
Comment (KaranGupta-git) [reply src/app/modules/auth/components/reset-password/reset-password.component.ts:36]: If a user will be redirected from mail link to the application his/her uid will be passed as URL param

**Pull request merged**